### ISSUE TITLE
pre-hook scripts run too late

### ISSUE THREAD

**Issue by ckastner** [certbot/certbot#8452]
Apparently, the pre-hook scripts are run after authentication happens, but authentication needs the network, so it is not possible to modify network settings through pre-hook scripts, for example modifying traffic filtering rules.

My firewall blocks everything by default, including outgoing. I attempted to use a pre-hook script `/etc/letsencrypt/renewal-hooks/pre/iptables` that simply opens up destination+port for `acme-v02.api.letsencrypt.org` and (its `-staging`) variant, but never got it to work. Neither did adding `--pre-hook`.

After some digging around, it appears that this is because the authenication part, through [init_le_client](https://github.com/certbot/certbot/blob/2a118f3e8331881aaecc132d5095b4941b7c710d/certbot/certbot/_internal/main.py#L581), happens before the hook scripts are run.

Would it be possible to execute the pre-hooks earlier? Alternatively, updating the documentation to this effect would also help.

Tested with certbot 1.8.0 on Debian 10.

Edit: this *should* be testable by blocking the servers, and then unblocking them with a script in `renewal-hooks`.

First, create the file /etc/letsencrypt/renewal-hooks/pre/iptables:
```shell
#!/bin/sh

iptables -D OUTPUT -p tcp -d acme-v02.api.letsencrypt.org,acme-staging-v02.api.letsencrypt.org -j DROP
```

Then, block the destination. This times out, because access to the servers is checked before the hook scripts are run:

```shell
$ sudo iptables -I OUTPUT 1 -p tcp -d acme-v02.api.letsencrypt.org,acme-staging-v02.api.letsencrypt.org -j DROP
$ certbot renew --force-renewal 
```

**Comment by bmw** [comment]
I think this is a duplicate of https://github.com/certbot/certbot/issues/5235.

I'm closing this issue in favor of that one, but please let me know if you think I missed something and they're different issues.

**Comment by ckastner** [comment]
It is indeed a duplicate -- sorry!
